lib/wasi_nn_rpc/wasi_ephemeral_nn.proto

```diff
@@ -18,6 +18,22 @@ enum TensorType {
   I64 = 6;
 }
 
+enum BackendError {
```

Comment:
Wondering if this should be grpc metadata
https://grpc.io/docs/guides/metadata/

Reply:
Using grpc metadata is indeed a good idea, but we should also return the backend error type to the client, because the client may behave differently depending on the error. For example, in the [llama-stream example](https://github.com/second-state/WasmEdge-WASINN-examples/blob/master/wasmedge-ggml/llama-stream/src/main.rs#L240-L252), the backend error type is used to determine whether to continue inference. If we use grpc metadata, we might also need to ensure that the client still has the opportunity to continue execution in the event of a backend error.

@hydai, do you have any thoughts?

Reply:
Please correct me if I am wrong.
The gprc is used between nnrpc-server and nnrpc-client. Once the client receives the grpc metadata, it will convert the metadata into a backend error and send it to the application.

In this case, we will need to:
1. Error occurs on the server side; wrap the backend error into grpc metadata
2. Server sends the grpc metadata to the client
3. The client decodes the grpc metadata and converts it back to the backend error
4. The client returns the backend error code to the WASM application

If we keep the backend error:
1. Error occurs on the server side; set the backend error as the error code.
2. The client receives the error code and returns it directly

I am not familiar with using grpc. Is it a best practice to use metadata to handle the errors? If so, we would love to adopt it.

Reply:
I think using metadata is fine, as the error should be supported for all RPC calls, not just `ComputeSingleResult`.

An alternative idea would be to just add `BackendError error = 1` to all the RPC calls without using metadata.